packages/trpc/server/routers/loggedInViewer/stripeCustomer.handler.ts

```diff
@@ -30,12 +30,29 @@ export const stripeCustomerHandler = async ({ ctx }: StripeCustomerOptions) => {
   }
 
   const metadata = userMetadata.parse(user.metadata);
-
-  if (!metadata?.stripeCustomerId) {
-    throw new TRPCError({ code: "BAD_REQUEST", message: "No stripe customer id" });
+  let stripeCustomerId = metadata?.stripeCustomerId;
+  if (!stripeCustomerId) {
+    // Create stripe customer
+    const customer = await stripe.customers.create({
+      metadata: {
+        userId: userId.toString(),
+      },
+    });
+    await prisma.user.update({
+      where: {
+        id: userId,
+      },
+      data: {
+        metadata: {
+          ...metadata,
+          stripeCustomerId: customer.id,
+        },
+      },
+    });
+    stripeCustomerId = customer.id;
```

Comment:
Little improvement, in the case user doesn't have stripeCustomerId in user metadata it should create one for him.

Reply:
In what situation would this happen?

Reply:
- An error during signup
- Not sure if getting invited would create you stripe customer id